### PR TITLE
Refactor variables.js

### PR DESCRIPTION
**Pull request by lannka**
Async expandTemplate is unnecessary.

### CODE REVIEW

File: extensions/amp-analytics/0.1/variables.js [hunk 1]
```diff
@@ -168,76 +169,53 @@ export class VariableService {
...
+  }
+
+  /**
+   * @param {string} template The template to expand
+   * @param {!ExpansionOptions} options configuration to use for expansion
+   * @return {string} The expanded string
+   */
+  expand(template, options) {
+    return template.replace(/\${([^}]*)}/g, (match, key) => {
+      if (options.iterations < 0) {
```
[the remark this answers is not in the document]
Comment (lannka) [reply extensions/amp-analytics/0.1/variables.js]: we print error only when there is unresolved variable.

File: extensions/amp-analytics/0.1/test/test-variables.js [hunk 2]
```diff
@@ -17,119 +17,136 @@
...
+        '1': '1${2}', '2': '2${3}', '3': '3${4}', '4': '4${1}',
+      };
+
+      it('defaults to 2 recursion', () => {
+        allowConsoleError(() => {
+          check('${1}', '123%24%7B4%7D', recursiveVars);
+        });
+      });
+
+      it('limits the recursion to 5', () => {
```
[the remark this answers is not in the document]
Comment (lannka) [reply extensions/amp-analytics/0.1/test/test-variables.js]: done

**Comment by zhouyx** [comment]
Can I know how you test between the two approaches. Also will DFS w/o promise block ui thread for too long? 
Otherwise the refactor looks good to me.

**Comment by zhouyx** [comment]
Did you run any performance test though?

**Pull request merged**